### PR TITLE
HubschAPI - adding public surface for hubs

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: include/signalrclient/hub_proxy.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        explicit hub_proxy(const std::shared_ptr<internal_hub_proxy>& proxy);
+        SIGNALRCLIENT_API ~hub_proxy();
+
+        SIGNALRCLIENT_API utility::string_t get_hub_name() const;
+
+        SIGNALRCLIENT_API void on(const utility::string_t& event_name, const std::function<void(const web::json::value &)>& handler);
+
+        template<typename T>
+        pplx::task<T> invoke(const utility::string_t& method_name,
+            const std::function<void(const web::json::value&)>& on_progress = [](const web::json::value&){})
```
Comment (jordanrudd) [inline include/signalrclient/hub_proxy.h]: What's the benefit of calling this method with the default for on_progress?

**Comment by jordanrudd** [comment]
Looks good. What's the status on the hub implementation? Currently I'm working on integrating our code with the connection class. Are you close enough that I should stop that work and prepare for taking a drop with hub support?

**Comment by moozzyk** [comment]
Once this is merged hubs work end-to-end. I actually have a test app that uses this right now as it stands. There are some nits and TODOs in the code but they just small things that should not affect using end ton end scenarios or edge cases (and bugs I don't know about yet).

**Comment by jordanrudd** [comment]
Ok, I’ll hold off on the fake-hub-around-connection then. Are you merging today?

Also, do you plan on supporting arm in the future?

**Comment by moozzyk** [comment]
I am planning all unmerged PRs but reviewed PRs today.

**Comment by moozzyk** [comment]
Thanks @jordanrudd !
